src/main/native/smithwaterman/PairWiseSW.h

```diff
@@ -417,7 +423,8 @@ void inline getCIGAR(SeqPair *p, int16_t *cigarBuf_, int32_t tid)
                    curSize += snprintf(p->cigar + curSize, expectedLength + 1, "%d%c", cigarArray[2 * i + 1], state);
            }
     }
-    p->cigarCount = strnlen(p->cigar, curSize);
+    // The length of the string can be truncated to int16_t because it will never be larger than the at most 16 bit long input.
+    p->cigarCount = (int16_t)strnlen(p->cigar, curSize);
```

Comment:
Note to me.  Review.

Reply:
Last issue to review/resolve.

Reply:
According to SAM file spect this should be uint32. https://samtools.github.io/hts-specs/SAMv1.pdf  section 4.2 pg 15. uint32 t[n cigar op]